core/domain/takeout_commit_message_truncate_jobs_one_off.py

```diff
@@ -0,0 +1,130 @@
+            exploration_models.ExplorationRightsSnapshotMetadataModel,
+            exploration_models.ExplorationSnapshotMetadataModel,
+            skill_models.SkillSnapshotMetadataModel,
+            story_models.StorySnapshotMetadataModel,
+            subtopic_models.SubtopicPageSnapshotMetadataModel,
+            topic_models.TopicRightsSnapshotMetadataModel,
+            topic_models.TopicSnapshotMetadataModel,
+            question_models.QuestionSnapshotMetadataModel,
+        ]
+
+    @staticmethod
+    def map(item):
+        model_name = item.__class__.__name__
+        model_id = item.id
+        identifier_message = '%s with id %s' % (model_name, model_id)
+        if len(item.commit_message) <= 1000:
+            yield ('LESS_OR_EQUAL_TO_1000', identifier_message)
+        elif len(item.commit_message) <= 1500:
+            yield ('BETWEEN_1000_AND_1500', identifier_message)
+        else:
+            yield ('GREATER_THAN_1500', identifier_message)
+
+    @staticmethod
+    def reduce(key, values):
+        if key == 'LESS_OR_EQUAL_TO_1000':
+            yield (key, len(values))
+        elif key == 'BETWEEN_1000_AND_1500':
+            yield (key, len(values))
+        else:
+            yield (key, values)
+
+
+class SnapshotMetadataCommitMsgShrinkOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that audits the commit_message field of the
+    BaseSnapshotMetadataModels to determine frequency of string sizes.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(SnapshotMetadataCommitMsgShrinkOneOffJob, cls).enqueue(
+            job_id, shard_count=4)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            config_models.ConfigPropertySnapshotMetadataModel,
+            collection_models.CollectionRightsSnapshotMetadataModel,
+            collection_models.CollectionSnapshotMetadataModel,
+            exploration_models.ExplorationRightsSnapshotMetadataModel,
+            exploration_models.ExplorationSnapshotMetadataModel,
+            skill_models.SkillSnapshotMetadataModel,
+            story_models.StorySnapshotMetadataModel,
+            subtopic_models.SubtopicPageSnapshotMetadataModel,
+            topic_models.TopicRightsSnapshotMetadataModel,
+            topic_models.TopicSnapshotMetadataModel,
+            question_models.QuestionSnapshotMetadataModel,
+        ]
+
+    @staticmethod
+    def map(item):
+        model_name = item.__class__.__name__
+        model_id = item.id
+        identifier_message = '%s with id %s' % (model_name, model_id)
+        if len(item.commit_message) > 1000:
+            item.commit_message = item.commit_message[:1000]
+            item.update_timestamps(update_last_updated_time=False)
+            item.put()
+            yield ('TRUNCATED', identifier_message)
+        else:
+            yield ('NOT_TRUNCATED', identifier_message)
```

Comment:
Just yield 1 as the second part of the tuple, since you don't end up using the message anyway.

Reply:
done